packages/flutter_tools/lib/src/commands/devices.dart

```diff
@@ -12,12 +12,34 @@ import '../globals.dart' as globals;
 import '../runner/flutter_command.dart';
 
 class DevicesCommand extends FlutterCommand {
+  DevicesCommand() {
+    argParser.addOption(
+      'timeout',
+      abbr: 't',
+      defaultsTo: null,
+      help: 'Time in seconds to wait for devices to attach. Longer timeouts may be necessary for networked devices.'
+    );
+  }
+
   @override
   final String name = 'devices';
 
   @override
   final String description = 'List all connected devices.';
 
+  Duration get timeout {
+    if (argResults['timeout'] == null) {
+      return null;
+    }
+    try {
+      final int timeoutSeconds = int.parse(stringArg('timeout'));
```

Comment:
use `int.tryParse` with a null check, then you only need to check for a nullable response